### PR TITLE
Converts Codebase to TypeScript

### PR DESCRIPTION
**Pull request by peetzweg**
Upgrades react and converts codebase to TypeScript.

### CODE REVIEW

**Comment by craftzdog** [comment]
Thanks but there are a lot of TS errors. Can you clean them up?

**Comment by peetzweg** [comment]
Added types throughout all files. ✌️

**Comment by craftzdog** [comment]
No, this commit [b30371b](https://github.com/craftzdog/chat-bubbles-for-yt/pull/10/commits/b30371b84001c33a7244b0d2f31fc49ed958bb19) broke the animations.

**Comment by peetzweg** [comment]
> No, this commit [b30371b](https://github.com/craftzdog/chat-bubbles-for-yt/pull/10/commits/b30371b84001c33a7244b0d2f31fc49ed958bb19) broke the animations.

Just pushed a fix. It was the missing `key` prop of the `<Bubble />`.
https://github.com/peetzweg/chat-bubbles-for-yt/blob/e6b7dd2a67d24f5986065971dcc3de5c8a4feb3c/src/App.tsx#L35

Weird though as component itself sets the `key` prop with the `id`. Guess my understanding of use of `key` in React is not yet complete. 🤔

File: src/use-messages.ts [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
+    (msg: Message) => {
       setMessages(messages => [...messages, msg])
       setTimeout(() => {
         setMessages(current => {
           const n = [...current]
           n.shift()
           return n
         })
-      }, 15000)
+      }, 3000)
```
Comment (craftzdog) [inline src/use-messages.ts:21]: Can you revert the timeout length to 15000?

**Review by craftzdog — CHANGES_REQUESTED** [review]

File: src/use-messages.ts (hunk 1 above)
Comment (peetzweg) [reply src/use-messages.ts]: Oh! Good catch. Had it changed for debugging the flickering issue. Not intended to change it.

**Comment by craftzdog** [comment]
Awesome! Thank you

**Pull request merged**